typescript/utils/src/is-empty.ts

```diff
@@ -0,0 +1,13 @@
+import { isEmpty as lodashIsEmpty } from "lodash/fp";
+
+/**
+ * Overrides Lodash `isEmpty` to assert that the value is also NonNullable
+ * @remarks
+ * This is permitted only because the Lodash `isEmpty` function also returns
+ * `true` if `value==null`.
+ * @param value - Value to test
+ * @returns True if the value is empty, false otherwise
+ */
+export const isEmpty = <TValue>(
```

Comment:
Good point! This function makes the lodash isEmpty function asserts that an array or string is null or undefined if it's empty. It was a messy way to avoid using `!isNil(someValue) && !isEmpty(someValue)` just to tell the TS compiler that the value can't be nil ; as it doesn't knows it when we only use `!isEmpty(someValue)` despite of having `!= null` in its implementation.
Considering that it's only used to cover the "value is not empty, nor nil" cases, I think that we should:
- [ ] rename this function `isNotEmpty`
- [ ] negate the return value of lodash `isEmpty`
- [ ] assert that `value is NonNullable<TValue>`

No?

Reply:
I'm still not sure that it should be used across the whole app as the function is covering a very specific need (the one in `typescript/web/src/components/core/button/button.tsx` file) related to the types and not to the behavior of the lodash function itself. 

In my opinion it would be better to just use the `!` [typescript operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) to deal with that problem:
in `typescript/web/src/components/core/button/button.tsx`, L28 is:

```ts
const iconProps = typeof icon === "string" ? { name: icon } : icon;
```

And would become:

```ts
const iconProps = typeof icon === "string" ? { name: icon! } : icon!;
```

Which would remove the need of the function and achieve the same goal